### ISSUE TITLE
Don't fail in the absence of `node_modules`

### ISSUE THREAD

**Issue by ericbottard** [cloudfoundry/npm-cnb#5]
Just because there is a package.json file doesn't mean there are dependencies. This is especially true for simple riff functions.

**Comment by kardolus** [comment]
I'm unable to reproduce this issue. I pushed this app to cflinuxfs3 `fixtures/empty_node_modules`. Both with an empty node_modules folder and without node_modules folder the app starts just fine.

Any chance you could provide us with a test app in order to reproduce this issue?

**Comment by kardolus** [comment]
Thanks for pointing this out! This [commit](https://github.com/cloudfoundry/npm-cnb/commit/3ec3fd5b07e375875c112e6067ab50f175bb239b) should fix your issue - Any chance you could give it a spin and tell us if this fixed it for you?

**Comment by ericbottard** [comment]
I'm not 100% sure if this is related to the particular latest version of this buildpack that I'm using (61ccaae9b60), but here is a weird error I'm getting (as part of an updated version of https://github.com/projectriff/riff-buildpack-group:

```
[builder] -----> NodeJS 11.4.0: Contributing to layer
[builder]        Downloading from https://nodejs.org/dist/v11.4.0/node-v11.4.0-linux-x64.tar.gz
[builder]        Verifying checksum
[builder]        Expanding to /workspace/org.cloudfoundry.buildpacks.nodejs/node
[builder]        Writing NODE_HOME to shared
[builder]        Writing NODE_ENV to shared
[builder]        Writing NODE_MODULES_CACHE to shared
[builder]        Writing NODE_VERBOSE to shared
[builder]        Writing NPM_CONFIG_PRODUCTION to shared
[builder]        Writing NPM_CONFIG_LOGLEVEL to shared
[builder]        Writing WEB_MEMORY to shared
[builder]        Writing WEB_CONCURRENCY to shared
[builder] failed to write launch.toml: mkdir /tmp/plan.917064464/org.cloudfoundry.buildpacks.npm/plan.toml: not a directory

```

**Comment by ericbottard** [comment]
Ran some additional tests from scratch and the above error is gone, please disregard.
However, I am indeed seeing this, which seems to imply that the issue this bug report is about is *not* fixed at 61ccaae:

```
[builder] -----> NPM Buildpack 0.0.3
[builder] -----> modules 98f1379ff60a0d3c6fa498326931498b75b6386773f59ac3e4e91b044710242a: Contributing to layer
[builder] Installing node_modules
[builder] up to date in 0.539s
[builder] found 0 vulnerabilities
[builder]
[builder] Cache verified and compressed (/workspace/app/npm-cache/_cacache):
[builder] Content verified: 0 (0 bytes)
[builder] Index entries: 0
[builder] Finished in 0.023s
[builder] unable to copy "/workspace/app/node_modules" to "/workspace/org.cloudfoundry.buildpacks.npm/modules": open /workspace/app/node_modules: no such file or directory
[builder] 2019/01/21 16:15:33 Error: failed to : exit status 103
ERROR: running builder in container: failed with status code: 7
```


For reference, here is the builder definition I'm using, where the relative path for the riff-buildpack is basically what can be found in [this PR](https://github.com/projectriff/riff-buildpack/pull/41):
```
[[buildpacks]]
id = "io.projectriff.riff"
#uri = "https://storage.googleapis.com/projectriff/riff-buildpack/latest.tgz"
uri = "../riff-buildpack/artifactory/io/projectriff/riff/io.projectriff.riff/latest/"

[[buildpacks]]
id = "org.cloudfoundry.buildsystem"
uri = "https://repo.spring.io/libs-milestone-local/org/cloudfoundry/buildsystem/org.cloudfoundry.buildsystem/1.0.0-M3/org.cloudfoundry.buildsystem-1.0.0-M3.tgz"

[[buildpacks]]
id = "org.cloudfoundry.openjdk"
uri = "https://repo.spring.io/libs-milestone-local/org/cloudfoundry/openjdk/org.cloudfoundry.openjdk/1.0.0-M3/org.cloudfoundry.openjdk-1.0.0-M3.tgz"

[[buildpacks]]
id = "org.cloudfoundry.buildpacks.nodejs"
uri = "https://github.com/cloudfoundry/nodejs-cnb/releases/download/v0.0.2/nodejs-cnb.tgz"

[[buildpacks]]
id = "org.cloudfoundry.buildpacks.npm"
uri = "https://storage.googleapis.com/projectriff/npm-buildpack/npm-cnb-61ccaae9b60.tgz"

[[groups]]

  [[groups.buildpacks]]
  id = "org.cloudfoundry.openjdk"
  version = "1.0.0-M3"
  optional = true # Irrelevant as this buildpack's detect is a noop, but set for clarity

  [[groups.buildpacks]]
  id = "org.cloudfoundry.buildpacks.nodejs"
  version = "0.0.2"
  optional = true # Irrelevant as this buildpack's detect is a noop, but set for clarity

  [[groups.buildpacks]]
  id = "org.cloudfoundry.buildsystem"
  version = "1.0.0-M3"
  optional = true

  [[groups.buildpacks]]
  id = "org.cloudfoundry.buildpacks.npm"
  version = "0.0.3"
  optional = true

  [[groups.buildpacks]]
  id = "io.projectriff.riff"
  version = "0.2.0-BUILD-SNAPSHOT"

```

**Comment by ericbottard** [comment]
Confirmed that 5c4c6ec2280 fixes this. 👍
